empress/clumpr/recon_vis/utils.py

```diff
@@ -0,0 +1,497 @@
+"""
+utils.py
+Utilities related to conversion between data types
+"""
+
+from typing import Dict, Tuple, List
+from recon import MappingNode, ReconGraph, Reconciliation
+from recon import Cospeciation, Duplication, Transfer, Loss, TipTip
+import tree
+from tree import NodeLayout
+from tree import TreeType
+
+__all__ = ['dict_to_tree', 'dict_to_reconciliation', 'build_trees_with_temporal_order']
+
+# Master utility function coverts from dictionaries to objects
+
+def convert_to_objects(host_dict, parasite_dict, recon_dict):
+    """
+    :param host_dict - dictionary representation of host tree
+    :param parasite_dict - dictionary representation of parasite tree
+    :param recon_dict - dictionary representation of reconciliation
+    :return - corresondoing host_tree and parasite_tree Tree objects
+        and recon Reconciliation object
+    """
+    host_tree, parasite_tree = build_trees_with_temporal_order(host_dict, parasite_dict, recon_dict)
+    recon = dict_to_reconciliation(recon_dict)
+    return host_tree, parasite_tree, recon
+
+# Tree utilities
+
+# Edge-based format is the primary format used by eMPRess algorithms.
+# This format comprises a dictionary in which each key is either the string
+# "hTop" ("pTop") for the edge corresponding to the handle of a host (parasite) tree
+# or an edge tuple of the form (v1, v2) where v1 and v2 are strings denoting the
+# name of the top and bottom vertices of that edge.  Values are 4-tuples of the form
+# (v1, v2, edge1, edge2) where edge1 and edge2 are the edge tuples for the
+# branches emanating from (v1, v2).  If the branch terminates at a leaf
+# then edge1 and edge2 are both None.
+# Here is an example of the host tree for the heliconius.newick file:
+
+host = {'hTop': ('Top', 'm1', ('m1', 'm2'), ('m1', 'm8')),
```

Comment:
Move this to a test file. If a user wants an example, he or she can refer to the test file.